Add type/status filters to tracking domains list

ghostwords opened the pull request:
Fixes #1422.

No mass controls yet (or #971, a way to reset settings), just new domain list filters. Is this good enough for now? Screenshots follow.

<details><summary>No filtering</summary>

![screenshot from 2017-12-11 17 22 53](https://user-images.githubusercontent.com/794578/33857060-3a3b2bfe-de98-11e7-83fd-bc648c78a6b8.png)
</details>

<details><summary>Show DNT-compliant only (disables status filter)</summary>

![screenshot from 2017-12-11 17 23 02](https://user-images.githubusercontent.com/794578/33857064-3d800d84-de98-11e7-90c4-c1b6196be96d.png)
</details>

<details><summary>Show user-controlled, partially-blocked only</summary>

![screenshot from 2017-12-11 17 23 37](https://user-images.githubusercontent.com/794578/33857066-416956a8-de98-11e7-9f9c-88257cc776e4.png)
</details>

alexristich commented:
I like the idea, though I think the UX needs some refinement. The two dropdowns don't have much distinction from each other.

Instead of dropdowns, perhaps radio buttons would work better here? Here's a very rough visual for comparison:
![radio_buttons](https://user-images.githubusercontent.com/11308076/34809945-3697434e-f64d-11e7-9d5d-10578b9fa96d.png)

ghostwords commented:
I am not sure radio buttons are called for in this situation. It seems like information overload to show all possible states at all times.

It might be more clear to break out the label of the control from control states though as you've done in the above mockup. Instead of "Show all DNT-compliant domains" with no label, we'd have "DNT-compliant domains" and a label of "Tracker Type".

ghostwords commented:
How about this?

![screenshot from 2018-01-16 17 45 21](https://user-images.githubusercontent.com/794578/35016107-3f040782-fae5-11e7-9a44-4f1a2a8a1044.png)

Responsive (mobile) mockup:

![screenshot from 2018-01-16 17 46 16](https://user-images.githubusercontent.com/794578/35016106-3cf58a9c-fae5-11e7-9ce2-0c00c5471cba.png)

alexristich commented on src/js/options.js:
```diff
@@ -273,21 +276,48 @@ function refreshOriginCache() {
+      if (status_filter != value.replace("user_", "")) {
+        return false;
+      }
+    }
```
[Style] Mix of `==` and `===` in some of these functions; do we have any preference for this?

ghostwords replied:
No strong preference ... I'll kick this particular can down the road for now.

alexristich commented on src/js/options.js:
```diff
@@ -417,14 +447,14 @@ function refreshFilterPage() {
-    originsToDisplay = allTrackingDomains;
-  }
-  showTrackingDomains(originsToDisplay);
+  // TODO when does this happen?
```
[Clarification] Is this an open question still?

ghostwords replied:
Just another note to self. Will resolve.

ghostwords commented on src/js/options.js:
```diff
@@ -46,7 +46,8 @@ let migrations = require("migrations").Migrations;
  */
 function loadOptions() {
-  $('#blockedResources').css('max-height',$(window).height() - 300);
+  // TODO blame
```
Just a note to self to check something (that doesn't seem to work at this point).

ghostwords commented on src/js/options.js:
```diff
@@ -273,21 +276,48 @@ function refreshOriginCache() {
 /**
  * Gets array of encountered origins.
+ *
  * @param filterText {String} Text to filter origins with.
```
Good call.

ghostwords commented:
Merged, added new string to Transifex (`tx push -s`, then manually downloaded and commited updated source locale: 2468014b1916d2e5dacf2e30fc342d19e119dbfb), updated all locales with new strings from Transifex (bf42e6f18fb79be2bf887b76adf4b92d79a2cd30).
